src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs

```diff
@@ -184,10 +184,11 @@ public void UpdateDescriptionForModel(MappingDataFormatParameter mappingSource)
             //The dimension is the first from the supported dimension which
             //has the selected unit.
             column.Unit = _mappingParameterEditorPresenter.Unit;
-            column.Dimension = _columnInfos
-               .First(x => x.DisplayName == model.MappingName)
-               .SupportedDimensions
-               .FirstOrDefault(x => x.HasUnit(column.Unit.SelectedUnit));
+            if (column.Dimension != null && !column.Dimension.HasUnit(column.Unit.SelectedUnit))
+               column.Dimension = _columnInfos
+                  .First(x => x.DisplayName == model.MappingName)
```

Comment:
it's in the place where you used the cache that you just implemented?
Also please add a { } around the if when what comes after is over multiple lines

Reply:
You are right. It was on the other PR but it was already merged so I am rebasing to include the changes